src/bin/sg-nvim-agent.rs

```diff
@@ -15,7 +17,16 @@ async fn main() -> Result<()> {
                 eprintln!("Recieved a message: {message:?}");
                 match message.respond().await {
                     Ok(response) => jsonrpc::write_msg(&mut stdout, response).await?,
-                    Err(err) => eprintln!("Failed to respond: {err:?}"),
+                    Err(err) => {
+                        jsonrpc::write_err(
+                            &mut stdout,
+                            RPCErr {
+                                code: 1,
+                                message: err.to_string(),
+                            },
+                        )
+                        .await?
+                    }
```

Comment:
should probably refactor this to just `write<T: Serialize>(... Result<T>)` or something